core/src/commonMain/kotlin/dev/fritz2/core/Lens.kt

```diff
@@ -178,6 +178,21 @@ fun <K, V> lensForElement(key: K): Lens<Map<K, V>, V> = object : Lens<Map<K, V>,
         else throw CollectionLensSetException("no item found with key='$key'")
 }
 
+/**
+ * create a [Lens] for upcasting a base (sealed) class or interface to a specific subtype.
+ */
+inline fun <P, reified C : P> lensForUpcasting(): Lens<P, C> = object : Lens<P, C> {
+    override val id: String = ""
+
+    override fun get(parent: P): C = try {
```

Comment:
I just had another thought on this in order to avoid the try-catch - would this be a viable alternative?

```kotlin
override fun get(parent: P): C = 
    (parent as? C) ?: throw CollectionLensGetException()
```